wikidatarefisland/pipes/item_extractor_pipe.py

```diff
@@ -0,0 +1,78 @@
+from itertools import chain
+from .abstract_pipe import AbstractPipe
+from wikidatarefisland.data_model import StatementFilterer
+from ..data_model.filters import StatementFilters
+
+
+class ItemExtractorPipe(AbstractPipe):
+    """Pipe that extracts statements that could be referenced from an item"""
+    def __init__(self, external_id_formatter,
+                 blacklisted_properties=None, whitelisted_ext_ids=None):
+        self.external_id_formatter = external_id_formatter
+        if blacklisted_properties is None:
+            blacklisted_properties = []
+        self.blacklisted_properties = blacklisted_properties
+        if whitelisted_ext_ids is None:
+            whitelisted_ext_ids = []
+        self.whitelisted_ext_ids = whitelisted_ext_ids
+
+    def flow(self, input_data):
+        return self._process_item(input_data)
+
+    def _process_item(self, item_data):
+        # TODO: Filter out bail early if item is in excluded list. see: T251282
+        all_statements = list(chain.from_iterable([i for i in item_data['claims'].values()]))
+        result_statements = list(self._filter_potential_referenced_statements(
+            all_statements))
+        resource_urls = list(self._extract_potential_resource_urls(all_statements))
+        return [{
+            'itemId': item_data['id'],
+            'statements': result_statements,
+            'resourceUrls': resource_urls
+        }]
```

Comment:
Generally speaking, is there a situation (apart from the one mentioned in the TODO comment above) where this pipe doesn't return an item? (lack of resource urls, lack of unreferenced statements, etc...)

Reply:
oops, just go an answer to my question (saw [T251944](https://phabricator.wikimedia.org/T251944) on the board) so nevermind :D